src/PowerShellEditorServices/Extensions/Api/EditorExtensionServiceProvider.cs

```diff
@@ -0,0 +1,215 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Linq.Expressions;
+using System.Reflection;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.PowerShell.EditorServices.Services;
+using Microsoft.PowerShell.EditorServices.Utility;
+using OmniSharp.Extensions.LanguageServer.Protocol.Server;
+
+using Internal = Microsoft.PowerShell.EditorServices.Services;
+
+namespace Microsoft.PowerShell.EditorServices.Extensions.Services
+{
+    /// <summary>
+    /// Object to provide extension service APIs to extensions to PSES.
+    /// </summary>
+    public class EditorExtensionServiceProvider
+    {
+        private static readonly Assembly s_psesAsm = typeof(EditorExtensionServiceProvider).Assembly;
+
+        private static readonly Lazy<object> s_psesAsmLoadContextLazy = new Lazy<object>(GetPsesAsmLoadContext);
+
+        private static readonly Lazy<Type> s_asmLoadContextType = new Lazy<Type>(() => Type.GetType("System.Runtime.Loader.AssemblyLoadContext"));
+
+        private static readonly Lazy<Func<IDisposable>> s_enterPsesReflectionContextLazy = new Lazy<Func<IDisposable>>(GetPsesAlcReflectionContextEntryFunc);
+
+        private static readonly Lazy<Func<string, Assembly>> s_loadAssemblyInPsesAlc = new Lazy<Func<string, Assembly>>(GetPsesAlcLoadAsmFunc);
+
+        private static Type AsmLoadContextType => s_asmLoadContextType.Value;
+
+        private static object PsesAssemblyLoadContext => s_psesAsmLoadContextLazy.Value;
+
+        private static Func<IDisposable> EnterPsesAlcReflectionContext => s_enterPsesReflectionContextLazy.Value;
+
+        private static Func<string, Assembly> LoadAssemblyInPsesAlc => s_loadAssemblyInPsesAlc.Value;
+
+        private readonly IServiceProvider _serviceProvider;
+
+        internal EditorExtensionServiceProvider(IServiceProvider serviceProvider)
+        {
+            _serviceProvider = serviceProvider;
+            LanguageServer = new LanguageServerService(_serviceProvider.GetService<ILanguageServer>());
+            //DocumentSymbols = new DocumentSymbolService(_serviceProvider.GetService<SymbolsService>());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4979332)